From 311521582be21285f1ce42fbdbac4579461e484a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ond=C5=99ej=20Jirman?= <megi@xff.cz>
Date: Tue, 16 Nov 2021 21:16:26 +0100
Subject: [PATCH 328/515] drm: dw-mipi-dsi-rockchip: Never allow lane bandwidth
 to be less than requested

Bandwidth can be less than requested in some cases, because the search
for best values only checked for absolute difference from ideal value.

This is likely not intentional.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 drivers/gpu/drm/rockchip/dw-mipi-dsi-rockchip.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/rockchip/dw-mipi-dsi-rockchip.c b/drivers/gpu/drm/rockchip/dw-mipi-dsi-rockchip.c
index 4e822333af05..b8b63a34d587 100644
--- a/drivers/gpu/drm/rockchip/dw-mipi-dsi-rockchip.c
+++ b/drivers/gpu/drm/rockchip/dw-mipi-dsi-rockchip.c
@@ -614,7 +614,7 @@ dw_mipi_dsi_get_lane_mbps(void *priv_data, const struct drm_display_mode *mode,
 			continue;
 
 		delta = abs(fout - tmp);
-		if (delta < min_delta) {
+		if (delta < min_delta && fout < tmp) {
 			best_prediv = _prediv;
 			best_fbdiv = _fbdiv;
 			min_delta = delta;
-- 
2.35.3

